### PR TITLE
Use go_router package for navigation

### PR DESCRIPTION
**Pull request by shockbytes**
### Overview

Use the `go_router` package for navigation. 

The navigation routes are encapsulated in the enum `DanteRoute` class. The main reason for doing  this, is that all routes are defined in one place. The risk of using wrong routes with corresponding 404 errors are mitigated.

We can only navigate to routes that are defined in this enum.

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -64,26 +73,68 @@ class DanteXApp extends ConsumerWidget {
...
-      home: FutureBuilder<bool>(
-        future: _launcher(ref),
-        builder: (context, snapshot) {
-          if (snapshot.hasData) {
-            return snapshot.data! ? const MainPage() : const LoginPage();
-          } else {
-            return const Material(child: CircularProgressIndicator.adaptive());
-          }
-        },
-      ),
```
Comment (shockbytes) [inline lib/main.dart:76]: I moved this logic into the `BootPage` as the `MaterialApp.router` does not support a `home` property.

File: lib/src/ui/boot_page.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        children: [
+          Image.asset(
+            'assets/logo/ic-launcher.jpg',
+            width: 96,
+            height: 96,
+          ),
+          const SizedBox(height: 24),
+          SizedBox(
+            width: MediaQuery.of(context).size.width * 0.4,
+            child: const LinearProgressIndicator(),
```
Comment (shockbytes) [inline lib/src/ui/boot_page.dart:44]: We'll have at least a nice loading screen in the beginning.

File: lib/main.dart (hunk 1 above)
Comment (lockieRichter) [reply lib/main.dart:76]: Yeah, the booting logic will get tidied up in the PR for the riverpod stuff.

File: lib/src/ui/boot_page.dart (hunk 2 above)
Comment (lockieRichter) [reply lib/src/ui/boot_page.dart:44]: yes, that's one thing I love about the old app is the loading page when you log in is so clean!

**Review by lockieRichter — APPROVED** [review]
Great work

**Pull request merged**